featurebyte/service/feature_preview.py

```diff
@@ -111,9 +168,24 @@ async def preview_target_or_feature(
         dict[str, Any]
             Dataframe converted to json string
         """
-        graph = feature_or_target_preview.graph
-        feature_node = graph.get_node_by_name(feature_or_target_preview.node_name)
-        operation_struction = feature_or_target_preview.graph.extract_operation_structure(
+        if feature_or_target_preview.object_id is not None:
+            document: Any
+            try:
+                document = await self.feature_service.get_document(
+                    feature_or_target_preview.object_id
+                )
+            except DocumentNotFoundError:
+                document = await self.target_service.get_document(
+                    feature_or_target_preview.object_id
+                )
```

Comment:
Would it make sense to add a type field to `feature_or_target_preview` so that we don't have to guess which collection the `object_id` is referring to? Also I think `id` is a more common naming convention for such a field than `object_id`.